rclcpp/include/rclcpp/create_subscription.hpp

```diff
@@ -81,6 +82,10 @@ create_subscription(
       options,
       *node_topics->get_node_base_interface()))
   {
+    if (options.topic_stats_options.publish_period <= std::chrono::milliseconds(0)) {
+      throw std::invalid_argument("topic statistics publish period must be greater than 0");
```

Comment:
i would notify the current setting as well.

Reply:
Seems reasonable to do this.

Reply:
Added the current publish period to the exception message.